src/model/ScheduleDay.cpp

```diff
@@ -176,42 +181,123 @@ namespace model {
         return 0.0;
       }
 
-      std::vector<double> values = this->values();          // these are already sorted
-      std::vector<openstudio::Time> times = this->times();  // these are already sorted
+      TimeSeries ts = this->timeSeries();
```

Comment:
So, I'm probably overthinking/bikeshedding but conceptually it pains me that we calculate everything for the entire day when we want one value.

I'm going to write the rationale for why I think that's still a good thing (and probably make it a comment for future ref)

* we're talking about max 60 timesteps * 24 hours = 1440 points, but realistically more often than not you'll have a timestep of 6 (our default) or 4 so 96 to 144 points. So not a lot of points
* We cache the timeSeries
* More often than not, the use case will be to get the values for an entire day, such as what openstudio-standards does with it (eg https://github.com/NREL/openstudio-standards/blob/682d3b2187d541fdeb91949e5bacbc6bc83952ff/lib/openstudio-standards/standards/Standards.ThermalZone.rb#L345)

```suggestion
      // We'll calculate the entire day when we request a single value but that's on purpose:
      // * We're talking about max 60 timesteps * 24 hours = 1440 points,
      //   but realistically more often than not you'll have a timestep of 6 (our default) or 4 so 96 to 144 points. So not a lot of points
      // * We cache the timeSeries, and,
      // * More often than not, the use case is to do it for the entire day anyways (eg: openstudio-standards to determine occupancy schedules)
      TimeSeries ts = this->timeSeries();
```